docs/conf.py

```diff
@@ -196,18 +196,26 @@
     ),
 ]
 
-
 # -- Extension configuration -------------------------------------------------
 
 # This is processed by Jinja2 and inserted before each notebook
+# We use internal readthedocs variables to get the git version if available. Note that this is undocumented, however it
+# is shown in our readthedocs build logs, and is generated from a template:
+# https://github.com/readthedocs/readthedocs.org/blob/master/readthedocs/doc_builder/templates/doc_builder/conf.py.tmpl
 nbsphinx_prolog = r"""
+{% if env.config.html_context.github_version is defined and env.config.html_context.current_version != "stable" %}
+    {% set github_version = env.config.html_context.github_version %}
```

Comment:
Maybe we could call this something a little more specific, like

```suggestion
    {% set git_revision = env.config.html_context.github_version %}
```

where a revision seems to be what git calls value that can be a commit-or-branch-or-... (e.g. https://git-scm.com/docs/git-rev-parse#_specifying_revisions).

(This is totally minor and doesn't need to change.)